src/treeview/treeview.component.ts

```diff
@@ -170,4 +170,13 @@ export class TreeViewComponent implements AfterViewInit, OnInit, OnDestroy {
 	public isProjected() {
 		return this.treeViewService.contentProjected;
 	}
+
+	private copyNode(node: Node): Node {
+		// making a recursive shallow copy to avoid performance issues when deeply cloning templateRefs if defined in the node
+		const copiedNode = Object.assign({}, node)
+		if (node.children) {
+		  copiedNode.children = node.children.map(child => this.copyNode(child))
+		}
+		return copiedNode
+	}
```

Comment:
This works, but any chance we can use cloneDeep from lodash? I believe we have the dependency as part of CCA

Reply:
@Akshat55 the answer is in the comment i left there in line 175...i have this solution locally in my project and i was using cloneDeep before, but when it comes to having angular template references in the object...then the cloneDeep is messing up with angular making it not refreshing correctly or extremely low performat...like i had huge wait times before the UI was refreshing